packages/langium/src/grammar/type-system/type-collector/types.ts

```diff
@@ -316,6 +329,27 @@ function isInterfaceAssignable(from: InterfaceType, to: InterfaceType, visited:
     return false;
 }
 
+function propertyTypeToKeyString(type: PropertyType): string {
+    if (isReferenceType(type)) {
+        return `@(${propertyTypeToKeyString(type.referenceType)})}`;
+    } else if (isArrayType(type)) {
+        return `(${propertyTypeToKeyString(type.elementType)})[]`;
+    } else if (isPropertyUnion(type)) {
+        const union = type.types.map(e => propertyTypeToKeyString(e)).join(' | ');
```

Comment:
These should be sorted so that type `X | Y` and `Y | X` get the same key (as they are the same type).

Reply:
Is that relevant in practice? It seems like an optimization for very rare cases, while in most other cases the sorting is an unnecessary step.